python-package/xgboost/core.py

```diff
@@ -862,7 +848,7 @@ def set_group(self, group):
         group : array like
             Group size of each group
         """
-        if _use_columnar_initializer(group):
+        if hasattr(group, "__cuda_array_interface__"):
```

Comment:
Does a cudf dataframe with single column have the array interface?  Also does a cudf series have array interface?  Can you be sure that the cudf test is using correct initializer?

Reply:
> Does a cudf dataframe with single column have the array interface?

If it does, then we need to make sure the cudf DataFrame is not initialized with `cupy` initializer.

Reply:
cudf with one column does not have array interface. I will modify it to accept this for metainfo.